### PR TITLE
[TIMOB-19578] Material design custom animations - touch feedback

### PR DESCRIPTION
**Pull request by sriks**
JIRA https://jira.appcelerator.org/browse/TIMOB-19578

### CODE REVIEW

File: apidoc/Titanium/UI/View.yml [hunk 1]
```diff
@@ -1063,6 +1063,23 @@ properties:
...
 
     
+  - name: touchFeedback
+    summary: A material design visual construct that provides an instantaneous visual confirmation of touch point.
+    description: |
+        This is an opt-in feature available from Android Lollipop. 
+        Touch feedback is applied only if the backgroundColor is a solid color.
+    type: Boolean
+    default: false
+    platforms: [android]
```
Comment (hieupham007) [inline apidoc/Titanium/UI/View.yml]: missing since: "5.2.0"

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 2]
```diff
@@ -1032,6 +1054,39 @@ private void applyCustomBackground(boolean reuseCurrentDrawable)
 			nativeView.setBackgroundDrawable(background);
 		}
 	}
+	
+	/**
+	 * @param props View's property dictionary
+	 * @return true if touch feedback can be applied. 
+	 */
+	protected boolean canApplyTouchFeedback(@NonNull KrollDict props) {
```
Comment (hieupham007) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: @Nonnull is Java 8 only right? Would this compile with Java 7?

File: apidoc/Titanium/UI/View.yml [hunk 3]
```diff
@@ -1063,6 +1063,23 @@ properties:
...
+
+    
+  - name: touchFeedbackColor
+    summary: Optional touch feedback ripple color. 
+    type: String
+    default: Theme provided color. 
+    platforms: [android]
```
Comment (hieupham007) [inline apidoc/Titanium/UI/View.yml]: missin since: "5.2.0"

File: apidoc/Titanium/UI/View.yml [hunk 4]
```diff
@@ -1063,6 +1063,23 @@ properties:
...
+    description: |
+        This is an opt-in feature available from Android Lollipop. 
+        Touch feedback is applied only if the backgroundColor is a solid color.
+    type: Boolean
+    default: false
+    platforms: [android]
+
+    
+  - name: touchFeedbackColor
+    summary: Optional touch feedback ripple color.
```
Comment (hieupham007) [inline apidoc/Titanium/UI/View.yml]: You should mention that this property does nothing unless `touchFeedback` is true

**Comment by hieupham007** [comment]
Please update with master and resolve conflicts

**Comment by sriks** [comment]
Updated with master and resolved conflicts.

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 5]
```diff
@@ -1056,6 +1078,39 @@ private void applyCustomBackground(boolean reuseCurrentDrawable)
...
+	}
+	
+	/**
+	 * Applies touch feedback. Should check canApplyTouchFeedback() before calling this. 
+	 * @param bgColor The background color of the view. 
+	 * @param rippleColor The ripple color.
+	 */
+	private void applyTouchFeedback(@NonNull Integer bgColor, @Nullable Integer rippleColor) {
+		Context appContext = TiApplication.getInstance();
+        if (rippleColor == null) {
```
Comment (hieupham007) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: looks like some indentation issue?

Comment (sriks) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: Addressed indentation issue. Please review.

**Comment by hansemannn** [comment]
@sriks @hieupham007 Can we merge this as all comments are addressed?

**Comment by hansemannn** [comment]
@sriks @ashcoding Ping. We should change the docs `since` to 5.4.0 and go ahead.

**Comment by hansemannn** [comment]
Closing in favor of https://github.com/appcelerator/titanium_mobile/pull/8176.

**Pull request closed without merging**